### ISSUE TITLE
Describe an explicit upgrade path going from a ~3.0 release to 4.0+

### ISSUE THREAD

**Issue by asachs01** [Graylog2/documentation#1091]
## Context

There are a number of users who are performing upgrades from pre-3.1 , but we don't describe the path to get from pre-3.1 to 3.3 or 4.X. Given that there are some significant changes at 3.1.3 (IIRC), we should document what the path looks like so that 1.) Users can avoid any pitfalls and 2.) Successfully upgrade their deployments.

## Expected Content

"If you're migrating from ~3.0, you should first upgrade to 3.1.X and from there, proceed to upgrade to later versions"...or something akin to that.

**Comment by emanteuf** [comment]
I agree, I have been unable to perform the upgrade from 3.3 to 4.0 following the instructions on the documentation site.  Luckily I took a snapshot of the virtual machine each time I've attempted to perform the upgrade.  Here's an example of someone who ran into the exact same problem I'm running into and there's no clear solution to the problem:  https://community.graylog.org/t/after-upgrading-from-3-3-to-4-im-getting-server-currently-unavailable/18222
